src/Analyzer.JsonRuleEngine.FunctionalTests/ScopeSelectionTests.cs

```diff
@@ -0,0 +1,254 @@
+            DisplayName = "No resource type, wildcard path resolves single path")]
+        [DataRow(null, "resources[*]",
+            "resources[0]", "resources[1]", "resources[2]", "resources[3]", "resources[4]",
+            DisplayName = "No resource type, wildcard path resolves multiple paths")]
+        [DataRow("Microsoft.Storage/storageAccounts", null,
+            // No scopes expected evaluated
+            DisplayName = "Resource type matches none, no path")]
+        [DataRow("Microsoft.Network/virtualNetworks", null,
+            "resources[0]",
+            DisplayName = "Resource type matches 1, no path")]
+        [DataRow("Microsoft.Compute/virtualMachines", null,
+            "resources[3]", "resources[4]",
+            DisplayName = "Resource type matches multiple, no path")]
+        [DataRow("Microsoft.Storage/storageAccounts", "name",
+            // No scopes expected evaluated
+            DisplayName = "Resource type matches none, path not resolved")]
+        [DataRow("Microsoft.Network/virtualNetworks", "properties.addressSpace",
+            null,
+            DisplayName = "Resource type matches 1, path not resolved")]
+        [DataRow("Microsoft.Network/virtualNetworks", "location",
+            "resources[0].location",
+            DisplayName = "Resource type matches 1, path resolved")]
+        [DataRow("Microsoft.Network/virtualNetworks", "dependsOn[*]",
+            // No scopes expected evaluated
+            DisplayName = "Resource type matches 1, wildcard path does not resolve")]
+        [DataRow("Microsoft.Network/virtualNetworks", "properties.subnets[*]",
+            "resources[0].properties.subnets[0]",
+            DisplayName = "Resource type matches 1, wildcard path resolves single path")]
+        [DataRow("Microsoft.Network/virtualNetworks", "properties.*",
+            "resources[0].properties.subnets[0]", "resources[0].properties.enableDdosProtection",
+            DisplayName = "Resource type matches 1, wildcard path resolves multiple paths")]
+        [DataRow("Microsoft.Network/networkInterfaces", "properties.dnsSettings",
+            null, null,
+            DisplayName = "Resource type matches multiple, path not resolved")]
+        [DataRow("Microsoft.Network/networkInterfaces", "properties.networkSecurityGroup",
+            "resources[1].properties.networkSecurityGroup", null,
+            DisplayName = "Resource type matches multiple, path resolves in 1")]
+        [DataRow("Microsoft.Network/networkInterfaces", "properties.ipConfigurations[0]",
+            "resources[1].properties.ipConfigurations[0]", "resources[2].properties.ipConfigurations[0]",
+            DisplayName = "Resource type matches multiple, path resolves in all")]
+        [DataRow("Microsoft.Compute/virtualMachines", "properties.hardwareProfile.*",
+            // No scopes expected evaluated
+            DisplayName = "Resource type matches multiple, wildcard path does not resolve")]
```

Comment:
And is this resource type correct?

Reply:
Yes - what's the reason you're unsure?  Is it the path not matching anything in the resources?  It's not supposed to.